File: database/src/main/java/org/difft/app/database/models/DraftModel.java
package org.difft.app.database.models;

import com.tencent.wcdb.WCDBField;
import com.tencent.wcdb.WCDBIndex;
import com.tencent.wcdb.WCDBTableCoding;

import java.util.Objects;

@WCDBTableCoding
public class DraftModel {

    @WCDBField(isPrimary = true, isUnique = true)
    public String roomId;

    @WCDBField
    public String draftJson;

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        DraftModel that = (DraftModel) o;
        return Objects.equals(roomId, that.roomId) && Objects.equals(draftJson, that.draftJson);
    }

    @Override
    public int hashCode() {
        return Objects.hash(roomId, draftJson);
    }
}

File: database/src/main/java/org/difft/app/database/models/DraftModel.kt
package org.difft.app.database.models

import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBTableCoding

@WCDBTableCoding
class DraftModel {

    @WCDBField(isPrimary = true, isUnique = true)
    var roomId: String? = null

    @WCDBField
    var draftJson: String? = null

    @WCDBField
    var updatedAt: Long = 0

    override fun equals(other: Any?): Boolean {
        if (other == null || javaClass != other.javaClass) return false
        other as DraftModel
        return updatedAt == other.updatedAt && roomId == other.roomId && draftJson == other.draftJson
    }

    override fun hashCode(): Int = java.util.Objects.hash(roomId, draftJson, updatedAt)
}
